src/passes/literalExpressionEvaluator/rationalLiteral.ts

```diff
@@ -137,6 +137,28 @@ export class RationalLiteral {
   toString(): string {
     return `${this.numerator}/${this.denominator}`;
   }
+
+  and(other: RationalLiteral): RationalLiteral {
+    return new RationalLiteral(
+      this.numerator & other.numerator,
+      this.denominator & other.denominator,
+    );
+  }
+  or(other: RationalLiteral): RationalLiteral {
+    return new RationalLiteral(
+      this.numerator | other.numerator,
+      this.denominator | other.denominator,
+    );
+  }
+  xor(other: RationalLiteral): RationalLiteral {
+    return new RationalLiteral(
+      this.numerator ^ other.numerator,
+      this.denominator ^ other.denominator,
+    );
+  }
+  not(): RationalLiteral {
+    return new RationalLiteral(~this.numerator, this.denominator);
+  }
```

Comment:
This functionality is already defined here as `bitwiseNegate`, so no need for this, it can be removed.

It is also a fine idea to add the "bitwise" prefix to these functions as in: `bitwiseAnd`, `bitwiseOr`, `bitwiseXor`; so it can be different from the logical operators